packages/transpiler/src/index.ts

```diff
@@ -0,0 +1,114 @@
+import { getContract, isContract, throwIfInvalidNode } from './solc/ast-utils';
+import { transpile } from './transpiler';
+import {
+  transformConstructor,
+  transformContractName,
+  appendDirective,
+  prependBaseClass,
+  purgeContracts,
+  transformParents,
+  fixImportDirectives,
+  purgeVarInits,
+} from './transformations/index';
+import { getInheritanceChain } from './solc/get-inheritance-chain';
+import { Artifact } from './solc/artifact';
+import { Transformation } from './transformation';
+
+export interface OutputFile {
+  fileName: string;
+  source: string;
+  path: string;
+  contracts: string[];
+}
+
+interface FileTran {
+  transformations: Transformation[];
+  source: string;
+}
+
+export function transpileContracts(contracts: string[], artifacts: Artifact[]): OutputFile[] {
+  // check that we have valid ast tree
+  for (const art of artifacts) {
+    throwIfInvalidNode(art.ast);
+  }
+
+  // create contract name | id to artifact map for quick access to artifacts
+  const contractsToArtifactsMap = artifacts.reduce<Record<string | number, Artifact>>((acc, art) => {
+    acc[art.contractName] = art;
+    const contract = getContract(art.ast, art.contractName);
+    acc[contract.id] = art;
+    return acc;
+  }, {});
+
+  // build a list of all contracts to transpile
+  const contractsWithInheritance = [
+    ...new Set(contracts.map(contract => getInheritanceChain(contract, contractsToArtifactsMap)).flat()),
+  ].filter(contract => {
+    const art = contractsToArtifactsMap[contract];
+    const contractNode = getContract(art.ast, contract);
+    return isContract(contractNode);
+  });
+
+  // build a array of transformations per Solidity file
+  const fileTrans = contractsWithInheritance.reduce<Record<string, FileTran>>((acc, contractName) => {
+    const art = contractsToArtifactsMap[contractName];
+
+    const source = art.source;
+
+    const contractNode = getContract(art.ast, contractName);
+
+    if (!acc[art.fileName]) {
+      const directive = `\nimport "@openzeppelin/upgrades/contracts/Initializable.sol";`;
+
+      acc[art.fileName] = {
+        transformations: [
+          appendDirective(art.ast, directive),
+          ...fixImportDirectives(art, artifacts, contractsWithInheritance),
+          ...purgeContracts(art.ast, contractsWithInheritance),
```

Comment:
>assuming these were not cleared in a previous run by a `transform-parents`

To be clear, my understanding is this would never happen because the transformations are all calculated before being actually applied.